network/debinterfaces/bridge_test.go

```diff
@@ -0,0 +1,326 @@
+    mtu 1500`
+
+	expected := `
+auto eth0
+iface eth0 inet manual
+    mtu 1500
+
+auto eth1
+iface eth1 inet manual
+    mtu 1500
+
+auto br-eth0
+iface br-eth0 inet static
+    bridge_ports eth0
+
+auto br-eth1
+iface br-eth1 inet static
+    address 192.168.1.254
+    gateway 192.168.1.1
+    netmask 255.255.255.0
+    dns-nameservers 8.8.8.8
+    dns-search ubuntu.com
+    dns-sortlist 192.168.1.1/24 10.245.168.0/21 192.168.1.0/24
+    bridge_ports eth1`
+	s.assertBridge(input, expected[1:], c, map[string]string{"eth0": "br-eth0", "eth1": "br-eth1"})
+}
+
+func (s *BridgeSuite) TestBridgeVLAN(c *gc.C) {
+	input := `
+auto eth0.2
+iface eth0.2 inet static
+    address 192.168.2.3/24
+    vlan-raw-device eth0
+    mtu 1500
+    vlan_id 2`
+
+	expected := `
+auto eth0.2
+iface eth0.2 inet manual
+    vlan-raw-device eth0
+    mtu 1500
+    vlan_id 2
+
+auto br-eth0.2
+iface br-eth0.2 inet static
+    address 192.168.2.3/24
+    bridge_ports eth0.2`
+	s.assertBridge(input, expected[1:], c, map[string]string{"eth0.2": "br-eth0.2"})
+}
+
+func (s *BridgeSuite) TestBridgeBond(c *gc.C) {
+	input := `
+auto bond0
+iface bond0 inet static
+    address 10.17.20.211/24
+    gateway 10.17.20.1
+    dns-nameservers 10.17.20.200
+    bond-slaves none
+    bond-mode active-backup
+    bond-xmit_hash_policy layer2
+    bond-miimon 100
+    mtu 1500
+    hwaddress 52:54:00:1c:f1:5b
+    bond-lacp_rate slow`
+
+	expected := `
+auto bond0
+iface bond0 inet manual
+    bond-slaves none
+    bond-mode active-backup
+    bond-xmit_hash_policy layer2
+    bond-miimon 100
+    mtu 1500
+    hwaddress 52:54:00:1c:f1:5b
+    bond-lacp_rate slow
+
+auto br-bond0
+iface br-bond0 inet static
+    address 10.17.20.211/24
+    gateway 10.17.20.1
+    dns-nameservers 10.17.20.200
+    hwaddress 52:54:00:1c:f1:5b
+    bridge_ports bond0`
+	s.assertBridge(input, expected[1:], c, map[string]string{"bond0": "br-bond0"})
+}
+
+func (s *BridgeSuite) TestBridgeNoIfacesDefined(c *gc.C) {
+	input := `
+mapping eth0
+    script /path/to/pcmcia-compat.sh
+    map home,*,*,*                  home
+    map work,*,*,00:11:22:33:44:55  work-wireless
+    map work,*,*,01:12:23:34:45:50  work-static`
+	s.assertBridge(input, input[1:], c, map[string]string{"eth0": "br-eth0"})
+}
+
+func (s *BridgeSuite) TestBridgeBondMaster(c *gc.C) {
+	input := `
+auto ens5
+iface ens5 inet manual
+    bond-lacp_rate slow
+    mtu 1500
+    bond-master bond0
+    bond-xmit_hash_policy layer2
+    bond-mode active-backup
+    bond-miimon 100`
+	s.assertBridge(input, input[1:], c, map[string]string{"eth0": "br-eth0"})
+}
+
+func (s *BridgeSuite) TestBridgeNoIfacesDefinedFromFile(c *gc.C) {
+	stanzas, err := debinterfaces.ParseSource("testdata/ifupdown-examples", nil, s.expander)
+	c.Assert(err, gc.IsNil)
+	input := format(stanzas)
+	s.assertBridge(input, input, c, map[string]string{"non-existent-interface": "non-existent-bridge"})
+}
+
+func (s *BridgeSuite) TestBridgeAlias(c *gc.C) {
+	input := `
```

Comment:
Don't quote me, but I believe this was all discovered/fixed by:

```commit ac6b6a3dc8cb6627409316dc3a4b1bee30fcc2ed
Merge: 5e850a4 9d88997
Author: Juju bot <jujubot@users.noreply.github.com>
Date:   Sat Jul 30 07:30:30 2016 +0100

    Merge pull request #5790 from freyes/bug1602716
    
    Alias interface set as an alias of the bridge
    
    When the intefaces file is regenerated alias interfaces are left as
    they are, this makes IPv6 aliases unusable.
    
    Example input:
    ```
    iface eth0:1 iface static
        address 192.168.122.2/24
    ```
    Example output:
    ```
    iface br-eth0:1 iface static
        address 192.168.122.2/24
    ```
    Fixes [LP:#1602716](https://bugs.launchpad.net/juju-core/+bug/1602716)
```